src/test/java/com/lambdaschool/foundation/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,303 @@
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private HelperFunctions helperFunctions;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private JwtDecoder jwtDecoder;
+
+    @MockBean
+    private UserService userService;
+
+    @MockBean
+    private CityService cityService;
+
+    List<User> userList;
+
+    @BeforeEach
+    void setUp() {
+        userList = new ArrayList<>();
+
+//            City c1 = new City("Washington, District of Columbia");
+//            c1.setPopulationdensityrating(20);
+//            c1.setSafteyratingscore(90);
+//            c1.setCostoflivingscore(15);
+//            c1.setAverageincome(110287d);
+//            c1.setAveragetemperature(100);
+//            c1.setLat(33.3367f);
+//            c1.setLon(-90.1234f);
+//
+//            City c2 = new City("Fort Lauderdale, Florida");
+//            c2.setPopulationdensityrating(43);
+//            c2.setSafteyratingscore(86);
+//            c2.setCostoflivingscore(12);
+//            c2.setAverageincome(39477d);
+//            c2.setAveragetemperature(70);
+//            c2.setLat(32.7673f);
+//            c2.setLon(-96.7776f);
+
+        User u1 = new User("Arthur");
+        u1.setUserid(1);
+        userList.add(u1);
+        User u2 = new User("James");
+        u2.setUserid(2);
+        userList.add(u2);
+
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    @Test
+    public void listAllUsers() throws Exception {
+        String apiUrl = "/users/users";
+
+        Mockito.when(userService.findAll())
+            .thenReturn(userList);
+
+        RequestBuilder rb = MockMvcRequestBuilders.get(apiUrl)
+            .accept(MediaType.APPLICATION_JSON);
+
+        // the following actually performs a real controller call
+        MvcResult r = mockMvc.perform(rb)
+            .andReturn(); // this could throw an exception
+        String tr = r.getResponse()
+            .getContentAsString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String er = mapper.writeValueAsString(userList);
+
+        System.out.println("Expect: " + er);
+        System.out.println("Actual: " + tr);
+
+        assertEquals(er,
+            tr);
+    }
+
+    @Test
+    void getUserById() throws
+                       Exception {
+        String apiUrl = "/users/user/1";
+
+        Mockito.when(userService.findUserById(1))
+            .thenReturn(userList.get(1));
+
+        RequestBuilder rb = MockMvcRequestBuilders.get(apiUrl)
+            .accept(MediaType.APPLICATION_JSON);
+        MvcResult r = mockMvc.perform(rb)
+            .andReturn(); // this could throw an exception
+        String tr = r.getResponse()
+            .getContentAsString();
+
+        ObjectMapper mapper = new ObjectMapper();
+        String er = mapper.writeValueAsString(userList.get(1));
+
+        System.out.println("Expect: " + er);
```

Comment:
Console messages are not needed in test. You can pass a third parameter into the assert method for a custom console output. JUnit by default will print the difference to the console if the two parameters in assert() don't match

Reply:
Same comment for the rest of the test with System.out.println().
Example -
```
assertEquals("Test string", stringVar, "custom log message");
```

Reply:
Updated with my message in the assertEquals call. It looks like junit 5 has that message after the arguments whereas junit 4 has it before them.